### PR TITLE
Change scope of `test_constants.py`

### LINKED ISSUE #14: Test explicitly that constants are Python scalars
data-apis/array-api#169 clarified that the constants `e`, `inf`, `nan`, and `pi` should be Python scalars and not just behave like one. Our current tests

https://github.com/data-apis/array-api-tests/blob/6e3684b7948564b3ecd69f5b066aaf9c722120ec/array_api_tests/test_constants.py#L5-L11

don't account for that.

### PR DESCRIPTION
**Pull request by honno**
This PR:
- Removes dependencies on most `xp` functions, as they're not interesting IMO so unnecessarily increase dependencies
  - Note these tests were written before data-apis/array-api#169 where such use of arrays was necessary
  - Opted to use pure Python equalities instead
- Instance checks the constants should work like floats i.e. supports `__float__` 
  - `float` check would hurt duck typed implementations, see https://github.com/data-apis/array-api/issues/154#issuecomment-833746576
  - [`numerary`](https://posita.github.io/numerary/0.1/) stuff doesn't look necessary here (interesting for other parts of the suite tho)
- Adds interesting equality checks that @pmeier wrote for #15... useful if we're not assuming `float`

Clarifying we don't need constants to be specifically `float` should close #14 and #15.

### CODE REVIEW

File: array_api_tests/test_constants.py [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
+
+
+def assert_0d_float(name: str, x: Array):
+    assert dh.is_float_dtype(
+        x.dtype
+    ), f"xp.asarray(xp.{name})={x!r}, but should have float dtype"
+
+
+@pytest.mark.parametrize("name, n", [("e", math.e), ("pi", math.pi)])
+def test_irrational(name, n):
```
Comment (asmeurer) [inline array_api_tests/test_constants.py:22]: This test is actually testing that n is fractional, not irrational (floating point numbers aren't strictly "irrational", and it's impossible to test if they "represent" one).

File: array_api_tests/test_constants.py [hunk 2]
```diff
@@ -1,34 +1,51 @@
...
+    assert isinstance(c, SupportsFloat), f"{name}={c!r} does not look like a float"
+
+
+def assert_0d_float(name: str, x: Array):
+    assert dh.is_float_dtype(
+        x.dtype
+    ), f"xp.asarray(xp.{name})={x!r}, but should have float dtype"
+
+
+@pytest.mark.parametrize("name, n", [("e", math.e), ("pi", math.pi)])
```
Comment (asmeurer) [inline array_api_tests/test_constants.py:21]: Shouldn't this be xp.e and xp.pi? Although why not just parameterize `name` and lookup `getattr(xp, name)` in the test?

File: array_api_tests/test_constants.py (hunk 1 above)
Comment (asmeurer) [reply array_api_tests/test_constants.py:22]: I guess the "irrational" here means it's testing the irrational numbers. A bit confusing. Maybe `test_irrational_numbers` would be clearer.

**Comment by asmeurer** [comment]
Just a few comments. Looks good otheriwse.

File: array_api_tests/test_constants.py (hunk 2 above)
Comment (honno) [reply array_api_tests/test_constants.py:21]: Ah no, coz I'm using Python's `math` just to get the rough approximations (i.e. to test the `xp` constants are at least in-between the floor and ceil of the respective `math` constants).

**Pull request merged**